### PR TITLE
Add tensorflow EfficientDet example

### PR DESCRIPTION
**Pull request by Riyria-was-taken**
Signed-off-by: Jagoda Kamińska jk406179@students.mimuw.edu.pl

#### Why we need this PR?
*Pick one, remove the rest*
- It adds EfficientDet tensorflow example

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     Simplification and adjustments for DALI pipeline of https://github.com/google/automl.
 - Affected modules and functionalities:
     New files in docs/examples/use_cases/tensorflow
 - Key points relevant for the review:
     NA
 - Validation and testing:
     NA
 - Documentation (including examples):
     NA

**JIRA TASK**: NA

### CODE REVIEW

File: docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/fn_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+import math
+
+from absl import logging
+from glob import glob
+from pipeline import anchors
+
+from . import ops
+
+
+class EfficientDetPipeline:
```
Comment (awolant) [inline docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/fn_pipeline.py:27]: DALI pipeline

File: docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/ops.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import nvidia.dali as dali
+
+
```
Comment (awolant) [inline docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/ops.py:17]: DALI pipeline

File: docs/examples/use_cases/tensorflow/efficientdet/utils.py [hunk 3]
```diff
@@ -30,6 +28,82 @@
...
+        ).get_dataset(total_batch_size)
+    elif strategy:
+        from pipeline.dali.fn_pipeline import EfficientDetPipeline
+
+        if pipeline == PipelineType.dali_cpu:
+            raise ValueError(
+                "dali_cpu pipeline is not compatible with mulit_gpu mode :<"
+            )
+
+        def dali_dataset_fn(input_context):
```
Comment (awolant) [inline docs/examples/use_cases/tensorflow/efficientdet/utils.py:67]: Parallel TF.data

File: docs/examples/use_cases/tensorflow/efficientdet/utils.py [hunk 4]
```diff
@@ -30,6 +28,82 @@
...
+class PipelineType(Enum):
+    syntetic = 0
+    tensorflow = 1
+    dali_cpu = 2
+    dali_gpu = 3
+
+
+def dict_to_namedtuple(字典):
+    NamedTuple = namedtuple("NamedTuple", 字典.keys())
+    return NamedTuple._make(字典.values())
```
Comment (klecki) [inline docs/examples/use_cases/tensorflow/efficientdet/utils.py:40]: I would prefer to keep the variable names in English and limited to plain ascii. 
```suggestion
def dict_to_namedtuple(dict_instance):
    NamedTuple = namedtuple("NamedTuple", dict_instance.keys())
    return NamedTuple._make(dict_instance.values())
```

**Comment by awolant** [comment]
This PR was split into: #3115 and #3118

**Pull request closed without merging**